catalog/transformer/catalog-transformer-streaming-lib/src/main/java/org/codice/ddf/transformer/xml/streaming/lib/XMLInputTransformer.java

```diff
@@ -0,0 +1,318 @@
+     * Method to create a new {@link SaxEventHandlerDelegate}, configured to parse a metacard
+     * according to {@link XmlInputTransformer#saxEventHandlerConfiguration} and {@link XmlInputTransformer#metacardType}
+     *
+     * @return a new SaxEventHandlerDelegate
+     */
+    private SaxEventHandlerDelegate create() {
+
+        /*
+         * Gets new instances of each SaxEventHandler denoted in saxEventHandlerConfiguration
+         */
+        List<SaxEventHandler> filteredSaxEventHandlers = saxEventHandlerFactories.stream()
+                .filter(p -> saxEventHandlerConfiguration.contains(p.getId()))
+                .map(SaxEventHandlerFactory::getNewSaxEventHandler).collect(Collectors.toList());
+        /*
+         * Pass all the new handlers to configure and create a new SaxEventHandlerDelegate and sets
+         * the metacardType
+         */
+        return new SaxEventHandlerDelegate(filteredSaxEventHandlers).setMetacardType(metacardType);
+
+    }
+
+    /**
+     * Takes in an XML {@link InputStream} and returns a populated {@link Metacard}
+     * The Metacard is populated with all attributes that have been parsed by the {@link SaxEventHandler}s
+     * declared in {@link XmlInputTransformer#saxEventHandlerConfiguration}s
+     *
+     * @param inputStream an XML input stream to be turned into a Metacard
+     * @return a populated Metacard
+     * @throws CatalogTransformerException
+     * @throws IOException
+     */
+    public Metacard transform(InputStream inputStream)
+            throws CatalogTransformerException, IOException {
+        if (inputStream == null) {
+            throw new CatalogTransformerException();
+        }
+
+        /*
+         * Create the necessary new SaxEventHandlerDelegate
+         */
+        SaxEventHandlerDelegate delegate = create();
+        OutputStream outputStream = new ByteArrayOutputStream();
+
+        /*
+         * Split the input stream, so that we can use it for parsing as well as read it into the Metacard.METADATA attribute
+         */
+        TeeInputStream teeInputStream = delegate.getMetadataStream(inputStream, outputStream);
+
+        /*
+         * Read the input stream into the metacard - where all the magic happens
+         */
+        Metacard metacard = delegate.read(teeInputStream);
+
+        /*
+         * Read the metadata from the split input stream and set it on the Metacard.METADATA attribute.
+         * However, if the metadata is null or empty, throw an exception - we can't return a metacard
+         * with no metadata
+         */
+        String metadata;
+        if ((metadata = outputStream.toString()) == null || metadata.isEmpty()) {
+            throw new CatalogTransformerException(
+                    "Metadata is empty from output stream. Could not properly parse metacard.");
+        }
+        metacard.setAttribute(new AttributeImpl(Metacard.METADATA, outputStream.toString()));
+        try {
+            inputStream.close();
```

Comment:
Is there a reason not to use the `try-with-resources` pattern here? Also, since you catch `IOException`, do still need to throw it on the method? Typically `InputTransformer`s handle any other exceptions internally, and throw/rethrow a `CatalogTransformerException` if necessary.